subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/keystore/TestKeyStore.groovy

```diff
@@ -32,12 +32,24 @@ class TestKeyStore {
         new TestKeyStore(rootDir)
     }
 
+    /*
+     clientStore/serverStore only contains self-signed certificates for embedded HTTPS server.
+     To make the client work with both embedded HTTPS server and real-world HTTPS server (e.g. Maven Central),
+     we need to merge JDK's cacerts into the custom keystore/truststore via:
+
+     keytool -importkeystore -srckeystore <JDK cacerts file location> -destkeystore <resource>/test-key-store/keyStore
+     keytool -importkeystore -srckeystore <JDK cacerts file location> -destkeystore <resource>/test-key-store/trustStore
+
+     Note that default password for JDK cacerts is "changeit".
+
+     The current keyStore-adoptopenjdk-8/trustStore-adoptopenjdk-8 is created from AdoptOpenJDK8 cacerts.
+     */
     private TestKeyStore(TestFile rootDir) {
         keyStore = rootDir.file("clientStore")
         trustStore = rootDir.file("serverStore")
 
-        copyCertFile("test-key-store/keyStore", keyStore)
-        copyCertFile("test-key-store/trustStore", trustStore)
+        copyCertFile("test-key-store/keyStore-adoptopenjdk-8", keyStore)
+        copyCertFile("test-key-store/trustStore-adoptopenjdk-8", trustStore)
```

Comment:
Can we now remove `test-key-store/trustStore`?

Reply:
No we can't in case someone wants to update the merged trust store.

Reply:
Ah, I didn't realize `trustStore` contains a key and `trustStore-adoptopenjdk-8` contains this one key + the keys from Adopt Openjdk 8.